features/FEATURE_LWIP/lwip-interface/ppp_lwip.cpp

```diff
@@ -0,0 +1,424 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2016 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <errno.h>
+#include "platform/FileHandle.h"
+#include "platform/mbed_poll.h"
+#include "events/EventQueue.h"
+#if defined(FEATURE_COMMON_PAL)
+#include "mbed_trace.h"
+#define TRACE_GROUP "LPPP"
+#else
+#define tr_debug(...) (void(0)) //dummies if feature common pal is not added
+#define tr_info(...)  (void(0)) //dummies if feature common pal is not added
+#define tr_error(...) (void(0)) //dummies if feature common pal is not added
+#endif //defined(FEATURE_COMMON_PAL)
+#include "rtos/Thread.h"
+#include "lwip/tcpip.h"
+#include "lwip/tcp.h"
+#include "lwip/ip.h"
+#include "lwip/dns.h"
+#include "lwip/pbuf.h"
+extern "C" { // "pppos.h" is missing extern C
+#include "netif/ppp/pppapi.h"
+}
+
+#include "nsapi_ppp.h"
+#include "ppp_lwip.h"
+#include "lwip_stack.h"
+
+namespace mbed {
+
+using rtos::Thread;
+using events::EventQueue;
+
+#if LWIP_PPP_API
+
+static EventQueue *event_queue;
+static Thread *event_thread;
+static volatile bool event_queued;
+static nsapi_error_t connect_error_code;
+
+// Just one interface for now
+static FileHandle *my_stream;
+static ppp_pcb *my_ppp_pcb;
+static bool ppp_link_up = false;
+static bool ppp_active = false;
+static const char *login;
+static const char *pwd;
+static sys_sem_t ppp_close_sem;
+static void (*notify_ppp_link_down)(nsapi_error_t) = 0;
+
+static EventQueue *prepare_event_queue()
+{
+    if (event_queue) {
+        return event_queue;
+    }
+
+    // Should be trying to get a global shared event queue here!
+    // Shouldn't have to be making a private thread!
+
+    // Only need to queue 1 event. new blows on failure.
+    event_queue = new EventQueue(2 * EVENTS_EVENT_SIZE, NULL);
+#if LWIP_DEBUG
+    event_thread = new Thread(osPriorityNormal, 900*2);
```

Comment:
That DEFAULT_THREAD_STACKSIZE is used to set the size of lwIP's main thread, and some drivers pick it up to use as the size of their own threads - I think it's debatable whether coupling those makes sense.

In this case, my recollection is that this PPP thread needed more stack than the lwip main thread, so it got set to that 900 independently. I could be wrong though.